core/src/main/java/org/teavm/backend/lowlevel/generate/LowLevelNameProvider.java

```diff
@@ -191,10 +191,15 @@ private String sanitize(String name) {
                 case '>':
                 case '<':
                 case '$':
+                case '_':
                     sb.append('_');
                     break;
                 default:
-                    sb.append(c);
+                    if ('a' <= c && c <= 'z' || 'A' <= c && c <= 'Z' || '0' <= c && c <= '9' && i != 0) {
+                        sb.append(c);
+                    } else {
+                        sb.append(String.format("U%04X", (int) c));
```

Comment:
I think it's ok just to append '_' character here. Also, I usually avoid using of String.format

Reply:
than `λ` and `σ` both will have `_` name which result in conflict

Reply:
This method is expected to return same names for different methods. Generation of non-conflicting names is handled in another place.